### PR TITLE
Update global random seed test_sparsefuncs

### PR DESCRIPTION
**Pull request by Tony-beeper**
**Reference Issues/PRs**

Fixes sklearn/utils/tests/test_sparsefuncs.py in https://github.com/scikit-learn/scikit-learn/issues/22827 upstream.

What does this implement/fix? Explain your changes.
This upgrades the test_sparsefuncs tests to account for greater randomness using global_random_seed.

**Any other comments?**
Most of the tests were able to pass after changing the global_random_seed variable. Only a few occasions methods such as increase sample size and reduce noise were needed. See comments on code changes for more details on these occasions.

### CODE REVIEW

**Review by ShoaibKhan — APPROVED** [review]
All test cases passed on my end. Your in-line comments are helpful to understand why some changes were made! LGTM 💯

**Review by tahmid-haque — APPROVED** [review]
Looks good! All tests pass.
